src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -3116,8 +3116,10 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
      * type arguments.
      */
     override def tpe_* : Type = {
-      maybeUpdateTypeCache()
-      tpeCache
+      this.synchronized {
```

Comment:
is this safe - can we cross symbols with this lock?
I dont see how, but there are checks for cycles, so if there are it implies crossing

Reply:
I think we need to discuss this.
Stack trace for the reference:

```
[WARN] Method accessed from multiple threads (scalac-parser-worker-3, scalac-parser-worker-2): scala.reflect.internal.Symbols$TypeSymbol.tpeCache_$eq(scala.reflect.internal.Types$Type)
    scala.reflect.internal.Symbols$TypeSymbol.tpeCache_$eq(Symbols.scala)
    scala.reflect.internal.Symbols$TypeSymbol.updateTypeCache(Symbols.scala:3155)
    scala.reflect.internal.Symbols$TypeSymbol.maybeUpdateTypeCache(Symbols.scala:3143)
    scala.reflect.internal.Symbols$TypeSymbol.tpe_$times(Symbols.scala:3120)
    scala.reflect.internal.Symbols$Symbol.tpe(Symbols.scala:1463)
    scala.tools.nsc.javac.JavaParsers$JavaParser.addAnnot$1(JavaParsers.scala:350)
    scala.tools.nsc.javac.JavaParsers$JavaParser.modifiers(JavaParsers.scala:386)
    scala.tools.nsc.javac.JavaParsers$JavaParser.typeBodyDecls(JavaParsers.scala:738)
    scala.tools.nsc.javac.JavaParsers$JavaParser.typeBody(JavaParsers.scala:728)
    scala.tools.nsc.javac.JavaParsers$JavaParser.classDecl(JavaParsers.scala:700)
    scala.tools.nsc.javac.JavaParsers$JavaParser.$anonfun$typeDecl$3(JavaParsers.scala:868)
    scala.tools.nsc.javac.JavaParsers$JavaParser$$Lambda$568/954745645.apply(Unknown Source)
    scala.tools.nsc.javac.JavaParsers$JavaParser.joinComment(JavaParsers.scala:138)
    scala.tools.nsc.javac.JavaParsers$JavaParser.typeDecl(JavaParsers.scala:868)
    scala.tools.nsc.javac.JavaParsers$JavaParser.compilationUnit(JavaParsers.scala:916)
    scala.tools.nsc.javac.JavaParsers$JavaParser.parse(JavaParsers.scala:49)
    scala.tools.nsc.ast.parser.SyntaxAnalyzer.scala$tools$nsc$ast$parser$SyntaxAnalyzer$$initialUnitBody(SyntaxAnalyzer.scala:85)
    scala.tools.nsc.ast.parser.SyntaxAnalyzer$ParserPhase.apply(SyntaxAnalyzer.scala:99)
    scala.tools.nsc.Global$GlobalPhase.scala$tools$nsc$Global$GlobalPhase$$processUnit(Global.scala:443)
    scala.tools.nsc.Global$GlobalPhase$$anonfun$1.$anonfun$applyOrElse$1(Global.scala:414)
    scala.tools.nsc.Global$GlobalPhase$$anonfun$1$$Lambda$427/1917083500.apply(Unknown Source)
    scala.concurrent.Future$.$anonfun$apply$1(Future.scala:608)
    scala.concurrent.Future$$$Lambda$428/1526009370.apply(Unknown Source)
    scala.util.Success.$anonfun$map$1(Try.scala:250)
    scala.util.Success$$Lambda$431/987128726.apply(Unknown Source)
    scala.util.Try$.apply(Try.scala:208)
    scala.util.Success.map(Try.scala:250)
    scala.concurrent.Future.$anonfun$map$1(Future.scala:244)
    scala.concurrent.Future$$Lambda$429/1226711664.apply(Unknown Source)
    scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
    scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
    scala.concurrent.impl.Promise$$Lambda$430/1319647571.apply(Unknown Source)
    scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    java.lang.Thread.run(Thread.java:745)
```